### ISSUE TITLE
scalafix:ok disables scalafix:off

### ISSUE THREAD

**Issue by fommil** [scalacenter/scalafix#693]
anecdotally...

```scala
  // scalafix:off DisableSyntax.keywords.null,DisableSyntax.keywords.var,DisableSyntax.keywords.while
```

followed by some code like

```scala
    var value: String = null
    val decodedValue = mutable.Set /* scalafix:ok */ [String]()
```

will fail the check on the `null`. If everything is moved into the `scalafix:off` it is ok.

**Comment by marcelocenerine** [comment]
this is probably caused by the same bug I fixed [here](https://github.com/scalacenter/scalafix/pull/620):  (search for "bugs"). See if it gets resolved with 0.6.0-M2

**Comment by marcelocenerine** [comment]
ok, I found out what is going wrong here. The following test reproduces the issue:

```scala
/*
rules = DisableSyntax
DisableSyntax.keywords = [
  var
  while
  null
]
*/
package test.escapeHatch

object Foo {

  // scalafix:off DisableSyntax.keywords.null,DisableSyntax.keywords.var,DisableSyntax.keywords.while
  val x = null // scalafix:ok
  var y = null

  while (true) {
    println("foo")
  }
}
```

when you add `// scalafix:ok DisableSyntax`, scalafix disable the given rule before the expression starts and enables it back again at the end of the expression. However, if you don't specify any rule, scalafix will then disable all rules before the expression and (incorrectly) enables everything at the end of the expression. It should enable only the rules that were disabled before the expression.

**Comment by olafurpg** [comment]
Nice job tracking down the root cause of this ticket. Would you be interested in taking on this issue @marcelocenerine ?

**Comment by marcelocenerine** [comment]
@olafurpg yeah, definitely

**Comment by olafurpg** [comment]
Fixed in #718
